bio_hansel/scripts/write_sequence.py

```diff
@@ -0,0 +1,141 @@
+        max_sequence_value = len(record_seq)
+        sequences = str(record_seq)
+        curr_chrom_df = curr_df[curr_df['CHROM'].str.match(
+            chromosome.strip())]
+        curr_chrom_df['POS'] = curr_chrom_df.index
+        ref_seqs = curr_chrom_df.POS.apply(
+            get_subsequences,
+            args=(sequences, sequence_length, max_sequence_value))
+        alt_seqs = ref_seqs.str.slice(
+            0, sequence_length) + curr_chrom_df.ALT + ref_seqs.str.slice(
+            sequence_length + 1, sequence_length + sequence_length + 1)
+
+        curr_chrom_df['ref_sequences'] = ref_seqs
+        curr_chrom_df['alt_sequences'] = alt_seqs
+        df_list.append(curr_chrom_df)
+    return df_list
+
+
+def write_sequences(output_directory: str, df_list: List[pd.DataFrame],
+                    schema_name: str, group: str) -> None:
+    """Writes the sequences associated with each SNV into the output schema file
+    Args:
+        output_directory: directory where the schema would be located as indicated by the user
+        df_list: list of DataFrames related for each particular chromosome included in the reference genome file 
+        schema_name: the name of the output schema file
+        group: the group membership of the current group of SNVs
+
+    Returns:
+         Creates schema file in the output directory specified by the user
+    """
+
+    with open(os.path.join(output_directory, f"{schema_name}.fasta"),
+              "a+") as file:
+        for curr_df in df_list:
+            for index, row in curr_df.iterrows():
+                ratio_value = row.iloc[3]
+                position = index
+                chromosome = row['CHROM']
+                reference_snv = row['ref_sequences']
+                alternate_snv = row['alt_sequences']
+                sequence_string = get_sequence_string(ratio_value, chromosome, position, group, reference_snv,
+                                                      alternate_snv)
+                file.write(sequence_string)
+
+
+def get_subsequences(position: int, seq: str, sequence_length: int,
+                     max_sequence_value: int) -> str:
+    """Get the sequences that are before are after the specified SNV
+    Args:
+        position: the position of the SNV based off of the reference genome
+        seq: the set of sequences from the reference genome
+        sequence_length: the amount of upstream and downstream sequences added to the SNV and included in the schema
+                        output
+        max_sequence_value: the length of the reference genome
+
+    Returns:
+        sequence_sequence: the sequence that is found at the specified genome positions from the reference genome
+
+    """
+    specific_sequence = seq[max(0, position - (sequence_length + 1)):min(
+        max_sequence_value, position + sequence_length)]
+    return specific_sequence
+
+
+def read_sequence_file(reference_genome_path: str) -> Dict[str, str]:
```

Comment:
The return looks like it's actually `Dict[str, Bio.Seq]` instead of what `Dict[str, str]`. How could you make it return the type you're specifying? Also PyCharm usually gives warnings about type mismatches in the right gutter. I find it useful to pay attention to what it's telling me the type actually is compared to what I'm saying it should be.

Reply:
yup this has been fixed in 1e8ecc07ceb58f25de3f4124ae887738c5ff5bef